rio_tiler/io/xarray.py

```diff
@@ -0,0 +1,362 @@
+
+            try:
+                dst_affine, w, h = self._dst_geom_in_tms_crs()
+
+                # The minzoom is defined by the resolution of the maximum theoretical overview level
+                # We assume `tilesize`` is the smallest overview size
+                overview_level = get_maximum_overview_level(w, h, minsize=tilesize)
+
+                # Get the resolution of the overview
+                resolution = max(abs(dst_affine[0]), abs(dst_affine[4]))
+                ovr_resolution = resolution * (2**overview_level)
+
+                # Find what TMS matrix match the overview resolution
+                self._minzoom = self.tms.zoom_for_res(ovr_resolution)
+
+            except:  # noqa
+                # if we can't get min/max zoom from the dataset we default to TMS maxzoom
+                warnings.warn(
+                    "Cannot determine maxzoom based on dataset information, will default to TMS maxzoom.",
+                    UserWarning,
+                )
+                self._minzoom = self.tms.maxzoom
+
+        return self._minzoom
+
+    def get_maxzoom(self) -> int:
+        """Define dataset maximum zoom level."""
+        if self._maxzoom is None:
+            # TODO: Get transform from the data and figure the MaxZoom
+            # self._minzoom = self.tms.zoom_for_res(resolution)
+            try:
+                dst_affine, _, _ = self._dst_geom_in_tms_crs()
+
+                # The maxzoom is defined by finding the minimum difference between
+                # the raster resolution and the zoom level resolution
+                resolution = max(abs(dst_affine[0]), abs(dst_affine[4]))
+                self._maxzoom = self.tms.zoom_for_res(resolution)
+
+            except:  # noqa
+                # if we can't get min/max zoom from the dataset we default to TMS maxzoom
+                warnings.warn(
+                    "Cannot determine maxzoom based on dataset information, will default to TMS maxzoom.",
+                    UserWarning,
+                )
+                self._maxzoom = self.tms.maxzoom
+
+        return self._maxzoom
+
+    @property
+    def minzoom(self):
+        """Return dataset minzoom."""
+        return self.get_minzoom()
+
+    @property
+    def maxzoom(self):
+        """Return dataset maxzoom."""
+        return self.get_maxzoom()
+
+    def info(self) -> Info:
+        """Return xarray.DataArray info."""
+        dims = [
+            d
+            for d in self.input.dims
+            if d not in ["lat", "lon", "longitude", "latitude", "x", "y"]
+        ]
+
+        bands = [str(band) for d in dims for band in self.input[d].values]
+        metadata = [band.attrs for d in dims for band in self.input[d]]
+
+        meta = {
+            "bounds": self.geographic_bounds,
+            "minzoom": self.minzoom,
+            "maxzoom": self.maxzoom,
+            "band_metadata": [(f"b{ix + 1}", v) for ix, v in enumerate(metadata)],
```

Comment:
Just an FYI, you can start enumerate at 1 with `enumerate(metadata, 1)` if you want :)